lms/djangoapps/instructor_task/api_helper.py

```diff
@@ -425,6 +425,16 @@ def encode_entrance_exam_and_student_input(usage_key, student=None):
     return task_input, task_key
 
 
+def encode_task_input(data):
+    """Encode optional data into task input."""
+    task_input = {}
+
+    if 'verified_learners_only' in data:
+        task_input['verified_learners_only'] = text_type(data.get('verified_learners_only')).lower() == 'true'
+
+    return task_input
+
+
```

Comment:
Why is the encode necessary? Can you elaborate the data mismatch, if any?

Reply:
Basically, We would like to convert 'true' / 'false' value in POST request to corresponding boolean. Once we have converted it, we are updating its str value to a boolean.